core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/dot-edit-content-host-folder-field.component.ts

```diff
@@ -0,0 +1,123 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    Input,
+    OnInit,
+    ViewChild,
+    inject,
+    signal
+} from '@angular/core';
+import { ControlContainer, FormControl, ReactiveFormsModule } from '@angular/forms';
+
+import { TreeSelect, TreeSelectModule } from 'primeng/treeselect';
+
+import { Site } from '@dotcms/dotcms-js';
+import { DotCMSContentTypeField } from '@dotcms/dotcms-models';
+
+import { DotEditContentFieldSingleSelectableDataTypes } from '../../models/dot-edit-content-field.type';
+import {
+    TreeNodeItem,
+    TreeNodeSelectEvent,
+    TreeNodeSelectItem
+} from '../../models/dot-edit-content-host-folder-field.interface';
+import { TruncatePathPipe } from '../../pipes/truncate-path.pipe';
+import { DotEditContentService } from '../../services/dot-edit-content.service';
+import { createPaths } from '../../utils/functions.util';
+
+@Component({
+    selector: 'dot-edit-content-host-folder-field',
+    standalone: true,
+    imports: [TreeSelectModule, ReactiveFormsModule, TruncatePathPipe],
+    templateUrl: './dot-edit-content-host-folder-field.component.html',
+    styleUrls: ['./dot-edit-content-host-folder-field.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    viewProviders: [
+        {
+            provide: ControlContainer,
+            useFactory: () => inject(ControlContainer, { skipSelf: true })
+        }
+    ]
+})
+export class DotEditContentHostFolderFieldComponent implements OnInit {
+    @Input() field!: DotCMSContentTypeField;
+    @ViewChild(TreeSelect) treeSelect!: TreeSelect;
+    readonly #controlContainer = inject(ControlContainer);
+    readonly #editContentService = inject(DotEditContentService);
+
+    options = signal<TreeNodeItem[]>([]);
+    pathControl = new FormControl();
+
+    ngOnInit() {
+        this.#editContentService.getSitesTreePath().subscribe((options) => {
+            this.options.set(options);
+            this.getInitialValue();
+        });
+    }
+
+    get formControl(): FormControl {
+        return this.#controlContainer.control.get(
+            this.field.variable
+        ) as FormControl<DotEditContentFieldSingleSelectableDataTypes>;
+    }
+
+    onNodeSelect(event: TreeNodeSelectEvent<Site>) {
+        const label = event.node.label;
+        if (label) {
```

Comment:
maybe guard clauses here and below to avoid nested ifs

if(!label) {
    return;
}